bucket/ddc_bucket/cluster/entity.rs

```diff
@@ -1,28 +1,31 @@
 //! The data structure of Clusters.
-
+// use ink_storage::Mapping;
+// use ink_prelude::vec::Vec;
 use ink_prelude::vec::Vec;
 use ink_storage::traits::{PackedLayout, SpreadLayout};
 use scale::{Decode, Encode};
 
-use crate::ddc_bucket::{AccountId, Balance, Error::InsufficientResources, NodeId, Result};
 use crate::ddc_bucket::cash::Cash;
-use crate::ddc_bucket::Error::UnauthorizedClusterManager;
-use crate::ddc_bucket::node::entity::{Node, Resource};
+use crate::ddc_bucket::node::entity::NodeId;
+use crate::ddc_bucket::node::entity::Resource;
 use crate::ddc_bucket::params::store::Params;
+use crate::ddc_bucket::Error::UnauthorizedClusterManager;
+use crate::ddc_bucket::{AccountId, Balance, Error::InsufficientResources, Result};
 
 pub type ClusterId = u32;
 pub type ClusterParams = Params;
 pub type VNodeIndex = u32;
 pub type VNodeId = (ClusterId, VNodeIndex);
 
-#[derive(Clone, PartialEq, Encode, Decode, SpreadLayout, PackedLayout)]
+#[derive(Clone, PartialEq, Encode, Decode, PackedLayout, SpreadLayout)]
 #[cfg_attr(feature = "std", derive(Debug, scale_info::TypeInfo))]
 pub struct Cluster {
     pub manager_id: AccountId,
-    pub vnodes: Vec<NodeId>,
     pub resource_per_vnode: Resource,
     pub resource_used: Resource,
     pub revenues: Cash,
+    pub node_ids: Vec<NodeId>,
+    pub v_nodes: Vec<Vec<u64>>,
```

Comment:
Why do we have a different type for v_nodes and what does it mean?

Reply:
because we have the ring topology and now there should be mapping of tokens which are placed on the ring  to the actual physical node ids. For instance, the first element of v_nodes (vector of v_nodes) will correspond to the first elment of node_ids